connector/src/main/java/org/jboss/as/connector/subsystems/resourceadapters/ResourceAdapterParser.java

```diff
@@ -321,7 +321,11 @@ private void parseResourceAdapter(final XMLExtendedStreamReader reader, final Li
                             }
                             String value = rawElementText(reader);
                             TRANSACTION_SUPPORT.parseAndSetParameter(value, operation, reader);
-                            isXa = value != null && TransactionSupportEnum.valueOf(value) == TransactionSupportEnum.XATransaction;
+                            ModelNode transactionSupport = TRANSACTION_SUPPORT.parse(value, reader);
```

Comment:
`SimpleAttributeDefinition#parse` is deprecated. Can you use `getParser` as JavaDoc suggests?